logger/logger.go

```diff
@@ -74,148 +69,77 @@ func CreateLogger(options Options, key string) (*Logger, error) {
 
 	options.setDefaults()
 	logger := Logger{
-		Key:      key,
-		Messages: make(chan Message),
-		Options:  options,
+		Options:   options,
+		transport: newTransport(options, key),
 	}
-	go logger.handleMessages()
+
 	return &logger, nil
 }
 
+// Close must be called when finished logging to ensure all buffered logs are sent
+func (l *Logger) Close() {
+	l.transport.close()
+}
+
 // Log sends a provided log message to LogDNA.
-func (logger *Logger) Log(message string) {
-	loggerMessage := Message{
+func (l *Logger) Log(message string) {
+	logMsg := Message{
 		Body:    message,
-		Options: logger.Options,
+		Options: l.Options,
 	}
-	logger.Messages <- loggerMessage
+	l.transport.add(logMsg)
 }
 
 // LogWithOptions allows the user to update options uniquely for a given log message
 // before sending the log to LogDNA.
-func (logger *Logger) LogWithOptions(message string, options Options) error {
-	msgopts := logger.Options.merge(options)
-	err := msgopts.validate()
+func (l *Logger) LogWithOptions(message string, options Options) error {
+	msgOpts := l.Options.merge(options)
+	err := msgOpts.validate()
 	if err != nil {
 		return err
 	}
 
-	loggerMessage := Message{
+	logMsg := Message{
 		Body:    message,
-		Options: msgopts,
+		Options: msgOpts,
 	}
 
-	logger.Messages <- loggerMessage
+	l.transport.add(logMsg)
 	return nil
 }
 
 // LogWithLevel sends a log message to LogDNA with a parameterized level.
-func (logger *Logger) LogWithLevel(message string, level string) error {
+func (l *Logger) LogWithLevel(message string, level string) error {
```

Comment:
why renaming this from `logger` to `l` since `logger` is more descriptive?

Reply:
this is a common convention in go for method
receiver variable names. there is some rationale for this here https://gist.github.com/adamveld12/c0d9f0d5f0e1fba1e551#receiver-names